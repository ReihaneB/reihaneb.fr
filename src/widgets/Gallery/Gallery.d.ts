import type { ImageProps } from '@/components/Image/Image.d';

interface ImageWithId extends ImageProps {
  id: number;
}

export interface GalleryProps {
  id?: number;
  name: string;
  description?: string;
  images: ImageWithId[];
}
