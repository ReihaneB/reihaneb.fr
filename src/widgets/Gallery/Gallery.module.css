.name {
  margin-bottom: 32px;
}

.description {
  margin-bottom: 32px;
}

.images {
  display: flex;
  flex-direction: column;
  gap: 32px;

  @media (width <= 768px) {
    gap: 16px;
  }
}