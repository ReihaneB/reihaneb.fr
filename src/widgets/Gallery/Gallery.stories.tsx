import type { Meta, StoryObj } from '@storybook/react';

import _16x9 from '@/styles/assets/placeholder/_16x9.png';
import _4x3 from '@/styles/assets/placeholder/_4x3.png';
import _1x1 from '@/styles/assets/placeholder/_1x1.png';

import Gallery from './Gallery';

const meta: Meta<typeof Gallery> = {
  title: 'Widgets/Gallery',
  component: Gallery,
};

export default meta;
type Story = StoryObj<typeof Image>;

const defaultArgs = {
  name: 'Gallery',
  images: [
    {
      id: 1,
      src: _16x9,
      alt: 'Placeholder image',
    },
  ],
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const SquareImage: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: _1x1,
        alt: 'Placeholder image',
      },
    ],
  },
};

export const _4x3Image: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: _4x3,
        alt: 'Placeholder image',
      },
    ],
  },
};

export const _16x9Image: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: _16x9,
        alt: 'Placeholder image',
      },
    ],
  },
};

export const SingleImage: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: _16x9,
        alt: 'Placeholder image',
      },
    ],
  },
};

export const MultipleImages: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: _16x9,
        alt: 'Placeholder image',
      },
      {
        id: 2,
        src: _16x9,
        alt: 'Placeholder image',
      },
      {
        id: 3,
        src: _16x9,
        alt: 'Placeholder image',
      },
    ],
  },
};

export const LongName: Story = {
  args: {
    ...defaultArgs,
    name: 'Gallery with a really long name that should not be truncated',
  },
};

export const MultipleFormats: Story = {
  name: 'Multiple formats (1:1, 4:3, 16:9)',
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: _1x1,
        alt: 'Placeholder image',
      },
      {
        id: 2,
        src: _4x3,
        alt: 'Placeholder image',
      },
      {
        id: 3,
        src: _16x9,
        alt: 'Placeholder image',
      },
    ],
  },
};

export const WithDescription: Story = {
  args: {
    ...defaultArgs,
    description: 'Description of the gallery',
  },
};
