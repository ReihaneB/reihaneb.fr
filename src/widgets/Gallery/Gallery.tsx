import Image from '@/components/Image/Image';

import type { GalleryProps } from './Gallery.d';
import styles from './Gallery.module.css';

function Gallery({
  name,
  description,
  images,
}: GalleryProps) {
  return (
    <>
      <h2 className={styles.name}>
        {name}
      </h2>
      {description && (
        <p className={styles.description}>
          {description}
        </p>
      )}
      <ul className={styles.images}>
        {images.map(({
          id,
          width,
          height,
          src,
          alt,
        }) => (
          <li key={id}>
            <Image
              width={width}
              height={height}
              src={src}
              alt={alt}
              priority
            />
          </li>
        ))}
      </ul>
    </>
  );
}

export default Gallery;
