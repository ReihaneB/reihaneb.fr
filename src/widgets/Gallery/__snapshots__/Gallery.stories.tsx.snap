// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/Gallery _4x3Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="800"
           height="600"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
           src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery _16x9Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery LongName smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery with a really long name that should not be truncated
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery MultipleFormats smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1080"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_1x1.ea48a62d.png?w=1080&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=3840&amp;q=75 2x"
           src="/static/media/_1x1.ea48a62d.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="800"
           height="600"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
           src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
           style="color: transparent;"
      >
    </li>
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery MultipleImages smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery SingleImage smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery SquareImage smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1080"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_1x1.ea48a62d.png?w=1080&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=3840&amp;q=75 2x"
           src="/static/media/_1x1.ea48a62d.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;

exports[`Widgets/Gallery WithDescription smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
    Gallery
  </h2>
  <p class="src-widgets-Gallery-Gallery-module__description--cwB1k">
    Description of the gallery
  </p>
  <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
    <li>
      <img alt="Placeholder image"
           fetchpriority="high"
           width="1920"
           height="1080"
           decoding="async"
           data-nimg="1"
           class="src-components-Image-Image-module__image--gXjYq"
           srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
           src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </li>
  </ul>
</main>
`;
